recipes/install.rb

```diff
@@ -100,8 +101,14 @@
       else
         "http://packages.treasuredata.com/#{major}/redhat/$releasever/$basearch"
       end
+    when '3'
+      if platform == "amazon"
+        amazon_version = node['platform_version'] == '2' ? '2' : '1'
```

Comment:
I'm not sure how the `platform_version` of future releases of Amazon Linux will be, but this kind of assumption will likely be problematic in some distant future.

How about to prepare some mapping of `platform_version` to Amazon Linux's release version as a `Hash` like follows...?

```rb
platform_version = node['platform_version']
amazon_version = node['td_agent']['yum_amazon_platform_version_mapping'][platform_version] || platform_version
```

To make the above working, we'd need to have some attribute definition like follows. With this way, users of cookbook can override the default mapping value if it is somehow insufficient.

```rb
default['td_agent']['yum_amazon_platform_version_mapping'] = {
  '2017.03' => '1',
  '2017.09' => '1',
  '2018.03' => '1',
  '2017.12' => '2',
}
```